src/sandstorm/run-bundle.c++

```diff
@@ -1270,6 +1313,23 @@ private:
       }
     }
 
+    // Now process the PORT setting, since the actual value in config.ports
+    // depends on if HTTPS_PORT was provided at any point in reading the
+    // config file.
+    //
+    // Outer KJ_IF_MAYBE so we only run this code if the config file contained
+    // a PORT= declaration.
+    KJ_IF_MAYBE(portValue, maybePortValue) {
+        KJ_IF_MAYBE(ports, getPorts(config.httpsPort, *portValue)) {
+            // Memory leak? Who owns this pointer? I guess it has to stay
+            // alive as long as the program does, since we don't typically
+            // delete the Config object.
+            config.ports = kj::mv(*ports);
+        } else {
+          KJ_FAIL_REQUIRE("invalid config value PORT", *portValue);
```

Comment:
Since you're throwing an exception anyway, you might consider making `getPorts()` itself throw the exception so that you don't have the ugliness of returning `Maybe`. But OTOH the error message might be not as nice. Up to you.